tests/unit/core/server/testServer.js

```diff
@@ -180,6 +195,12 @@ describe(__filename, () => {
         .end();
 
       expect(response.statusCode).toEqual(500);
+      mockUserApi.verify();
+
+      // This ensures the error caught is the expected one, that is the one
+      // from `userProfile` API call.
+      const { loadState } = store.getState().reduxAsyncConnect;
+      expect(loadState.ServerBase.error.sagaStack).toContain(errorMessage);
```

Comment:
is this assertion necessary? I would rather we didn't rely on `reduxAsyncConnect` because it will be removed in https://github.com/mozilla/addons-frontend/issues/2356

Reply:
I could not come with something better to be 100% sure the error caught came from the promise rejection, not something else (example: a check in the reducer called right after `callApi()` saying the `profile` argument was not provided).

I did that because I found a few fixes to make the test suite all green but that was not correct. For instance, it would return a `500` due to an error related to the reducer rather than the saga.

Reply:
This is a high-level integration test so I don't think it's crucial that we make an assertion about the exact error that triggered the 500. In other words, we won't rely on this specific test to tell us that fetching a user profile fully works as expected. We have better unit tests for that.

Reply:
Then, I can get rid of the promise "hack" I guess by not rejecting the promise but resolving it with empty/null data. The reducer will throw an error and this error will be caught, correctly.

Reply:
Well, we will need to set up a rejected promise in a server test sooner or later so setting a precedent for how to do it would be nice. I prefer tests that set up explicit errors rather than trying to create the error by side effect.

Reply:
If I understand correctly the problem (and remember what I have done last week), it is not because of the use of `Promise.reject()`. This, I am pretty sure, works and is handled somewhere in the code. I tend to think this side-effect is a false-positive.